cpp/include/cudf/io/functions.hpp

```diff
@@ -428,11 +433,149 @@ struct read_parquet_args {
   /// Cast timestamp columns to a specific type
   data_type timestamp_type{type_id::EMPTY};
 
+  explicit read_parquet_args(source_info const& src) : source(src) {}
+
+  friend read_parquet_args_builder;
+
+ public:
   explicit read_parquet_args() = default;
 
-  explicit read_parquet_args(source_info const& src) : source(src) {}
+  /**
+   * @brief create read_parquet_args_builder which will build read_parquet_args
+   *
+   * @param src source information used to read parquet file
+   * @returns read_parquet_args_builder
+   */
+  static read_parquet_args_builder build(source_info const& src);
+
+  /**
+   * @brief Returns source info
+   */
+  source_info get_source_info() const { return source; }
+
+  /**
+   * @brief Returns column names
+   */
+  std::vector<std::string> column_names() const { return columns; }
+
+  /**
+   * @brief Returns row groups
+   */
+  std::vector<std::vector<size_type>> get_row_groups() const { return row_groups; }
+
+  /**
+   * @brief Returns number of rows to skip
+   */
+  size_type rows_to_skip() const { return skip_rows; }
+
+  /**
+   * @brief Returns row groups
+   */
+  size_type get_num_rows() const { return num_rows; }
+
+  /**
+   * @brief Returns strings_to_categorical
+   */
+  bool is_strings_to_categorical() const { return strings_to_categorical; }
+
+  /**
+   * @brief Returns use_pandas_metadata
+   */
+  bool utilize_pandas_metadata() const { return use_pandas_metadata; }
+
+  /**
+   * @brief Returns timestamp type
+   */
+  data_type get_timestamp_type() const { return timestamp_type; }
 };
 
+class read_parquet_args_builder {
+  read_parquet_args args;
+
+ public:
+  explicit read_parquet_args_builder() = default;
+
+  explicit read_parquet_args_builder(source_info const& src) : args(src) {}
+
+  /**
+   * @brief Set column names which needs to be read
+   */
+  read_parquet_args_builder& with_column_names(std::vector<std::string> column_names)
+  {
+    args.columns = column_names;
```

Comment:
```suggestion
    args.columns = column_names;
```
```suggestion
    args.columns = std::move(column_names);
```

Reply:
Should the argument be && then?

Reply:
no, it's fine as-is. If an lvalue is passed it will copy to pass the parameter and move from the parameter. If an rvalue is passed, it will move-construct the parameter and then move from it again.